core/health/shell/src/main/java/org/opennms/core/health/shell/HealthCheckCommand.java

```diff
@@ -59,6 +59,12 @@
     @Option(name = "-t", description = "Maximum number of milliseconds to wait before failing when waiting for a check to complete (e.g. try to establish a JMS session.")
     public long timeout = 5L * 1000L;
 
+    @Option(name = "--local", description = "Specify that only local health checks with no external dependencies are executed")
+    public boolean localChecksOnly;
+
+    @Option(name = "--list", description = "List all the registered health checks")
+    public boolean listChecks;
+
```

Comment:
Can we also add an option to supply list of checks that can be ignored

Reply:
Could you define what you mean by "ignore"?

Currently the intent is for the "--local" to "ignore" checks that have dependencies external to container.

Reply:
I was thinking of supplying `ignoreHealthCheckList` but if current structure is not feasible, that's okay  https://github.com/OpenNMS/opennms/blob/8649eb68ad6d7cfd0de5a453376fa175bb44737e/core/health/impl/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L23
```suggestion

```

Reply:
Ok, I think I understand, the plan of "--list" is to also show if the check is local or not. In a way this would show which checks would be ignored when "--local" is used